client/src/pages/WeeklyTable.js

```diff
@@ -38,19 +39,44 @@ const getBookingsByRow = (bookings, rowsCount, week) => {
 const getBookingsByDesk = (bookings, week) => {
 	const bookingsByDayWithDesk = {};
 	week.forEach((day) => {
-		bookingsByDayWithDesk[day] = bookings.filter((booking) => booking.date === day && booking.desk !== null  );
+		bookingsByDayWithDesk[day] = bookings.filter(
+			(booking) => booking.date === day && booking.desk !== null
+		);
 	});
 
-	const bookingsByDesk = {}; 
-	desks.forEach((desk)=>{
-		bookingsByDesk[desk.name] = week.map((day) => bookingsByDayWithDesk[day].find((booking)=> booking.desk === desk.name));
+	const bookingsByDesk = {};
+	desks.forEach((desk) => {
+		bookingsByDesk[desk.name] = week.map((day) =>
+			bookingsByDayWithDesk[day].find((booking) => booking.desk === desk.name)
+		);
 	});
-		
+
 	return bookingsByDesk;
 };
 
-const WeeklyTable = ({ bookings, rowsCount, refreshBooking }) => {
+const countAvailableDesksForDay = (bookings, date, maxDesksForDay) => {
+	
+	const countBookingsByDay = bookings.filter(
+		(booking) => booking.date === date
+	).length;
+
+	const availableDesks = maxDesksForDay - countBookingsByDay;
+
+	if (countBookingsByDay >= 0 && countBookingsByDay < maxDesksForDay) {
```

Comment:
or... now that you have availableDesks, your if statement can look a lot simpler

```
if (availableDesks === 0) {
  //...
} else {
  //...
}
```

Reply:
Wow, I missed it!